File: backend/internal/domain/entity/invite.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type Invite struct {
	Id        uuid.UUID `json:"id" db:"id"`
	SpaceId   int       `json:"spaceId,omitempty" db:"space_id"`
	ExpiresAt time.Time `json:"expiresAt" db:"expires_at"`
	MaxUses   int       `json:"maxUses" db:"max_uses"`
	UsedCount int       `json:"usedCount" db:"used_count"`
	CreatedAt time.Time `json:"createdAt" db:"created_at"`
}

type CreateInviteDTO struct {
	SpaceId       int `json:"spaceId"`
	HoursToExpire int `json:"hoursToExpire"`
	MaxUses       int `json:"maxUses"`
}

type InviteIdDTO struct {
	InviteId uuid.UUID `json:"inviteId"`
}

File: frontend/src/types/invite.ts
export interface Invite {
    inviteId: string;
	usernameInSpace?: string;
}

export interface InviteDTO {
    id: string;
	expiresAt?: string;
}

export interface DeleteInviteDTO {
    inviteId: string;
}

export interface createInviteDTO {
    spaceId: number;
    hoursToExpire: number;
    maxUses: number;
}

export interface createInviteForm {
    hoursToExpire: number;
    maxUses: number;
}